jobs/credhub/templates/application_spring.yml.erb

```diff
@@ -23,12 +23,19 @@
     database = p('credhub.data_storage.database')
     username = p('credhub.data_storage.username')
     password = p('credhub.data_storage.password')
+    replication = p('credhub.data_storage.replication', nil)
+
+    if !replication.nil? && !replication.empty?
+      url = "jdbc:mariadb:#{replication}://#{host}:#{port}/#{database}?autoReconnect=true"
+    else
+      url = "jdbc:mariadb://#{host}:#{port}/#{database}?autoReconnect=true"
+    end
 
     properties['spring']['flyway']['locations'] << 'classpath:/db/migration/mysql'
     properties['spring']['datasource'] = {
       'username' => username,
       'password' => password,
-      'url' => "jdbc:mariadb://#{host}:#{port}/#{database}?autoReconnect=true",
```

Comment:
After investigation, it seems to us like the `autoReconnect=true` parameter from this URL is enough to connect to Aurora MySQL. 

We have noticed that if we take if out, we are not able to connect; but when we keep it, we are unable to reproduce the connection error. Without the double negative: CredHub connects and fully works with Aurora.

Please let us know if your experience is different.

Reply:
@churib @benjaminguttmann-avtq ^^^ We're confused about how it's failing for you because 'autoReconnect=true' is always enabled for CredHub. 

This is with mariadb 2.7.6. The behavior is different with newer mariadb releases, which we're also investigating.